packages/snap-toolbox/src/charsParams/charsParams.ts

```diff
@@ -15,9 +15,9 @@ export function charsParams(params: ParameterObject): number {
 					return length + keyLength + 1 + ('' + val).length;
 				}, 0)
 			);
-		} else {
+		} else if (typeof value == 'string' || typeof value == 'number') {
 			return count + keyLength + 1 + ('' + value).length;
-		}
+		} else return count + keyLength;
```

Comment:
Currently with this change objects are allowed to be passed in - but we will not fully traverse them. Instead, we would only get the key length and maybe a length for "[object object]" - which is likely to fall short of the actual number of characters.

It would be nice to make this function recursive to handle any depth of object, as that would get us a more accurate count of the "chars". Additional testing would be required to ensure that a change like that is working as intended.

Reply:
Let's move to a separate issue.

Reply:
ended up being simpler to add than expected.